### DISCUSSION TITLE
Webhooks on prisma 2

### DISCUSSION THREAD

**Discussion by AikoRamalho** [prisma/prisma#6121] [Q&A]
With prisma 1 we could create webhooks inside the prisma.yml file.

I'm wondering how to migrate this to p2. Any ideas? I've found nothing related on this project issues or docs.

**Comment by ryands17** [comment]
@AikoRamalho 👋 

I think you're talking about [hooks](https://v1.prisma.io/docs/1.34/prisma-cli-and-configuration/prisma-yml-5cy7/#reference) as Prisma 1 didn't have any such thing as webhooks.

If so, then could you explain what you currently have in your `hooks` section? That can be replaced with NPM scripts.

**Reply by AikoRamalho** [reply]
Uh, idk what is happening or if the docs are outdated, but here is the definition of the "prisma.yml" file:
https://github.com/prisma/prisma-json-schema/blob/master/src/schema.json

As you can see, there is a "subscription" definition type with a subtype called "webhooks".

I'm wondering how to migrate this.

**Reply by nikolasburk** [reply]
Hey Aiko 👋  if I recall correctly, `webhooks` were triggered by a GraphQL subscription and then invoked a serverless function. Prisma 2 doesn't have the concept of webhooks any more and you'd most likely implement an equivalent functionality manually in your GraphQL layer.

Can you maybe give a concrete example of a webhook that you'd like to migrate? Also, could you share what kind of GraphQL library you are using in your app (e.g. Nexus, TypeGraphQL or SDL-first)?

Did you otherwise already complete the upgrade process and the webhooks are the last piece that needs to be migrated or where are you in the upgrade process in general? 🙂

**Reply by AikoRamalho** [reply]
Oh, thank you for your answer!

Here is my handle-new-message webhook. Every time someone creates/updates a message, the subscription is trigged and calls a google function.

![image](https://user-images.githubusercontent.com/15454141/111458081-c67c2180-86f7-11eb-9483-cf58a7911bbd.png)

The subscription def (p1 and old nexus):

![image](https://user-images.githubusercontent.com/15454141/111458322-0f33da80-86f8-11eb-851b-1bd41a3e2f3a.png)

In this case, node = message object that was created or update. 

I'm using nexus btw.

The webhooks are the last piece that needs to be migrate in the p1 to p2 migration, but I still haven't finished migrating from old nexus to new nexus.

**Reply by nikolasburk** [reply]
So, I think the way how to handle this would be as follows:

1. manually set up a serverless function (the equivalent of the function that you call currently via `${self.custom.google-endpoint}/{env:HANDLE_MESSAGE}`)
1. manually send an HTTP request with the same payload from the subscription inside your `createMessage` and `updateMessage` resolvers to that serverless function

That way you should get the same behaviour as before 🙂 let me know if that helps or if you have any further questions! 🙌

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
